pkg/reconciler/common/prune.go

```diff
@@ -170,5 +301,39 @@ func checkAndDelete(k kubernetes.Interface, ctx context.Context, targetNamespace
 	}
 
 	//if there is no error it means cron is exists, but no prune in config it means delete it
-	return k.BatchV1beta1().CronJobs(targetNamespace).Delete(ctx, CronName, v1.DeleteOptions{})
+	return pruner.kc.BatchV1beta1().CronJobs(ns).Delete(pruner.context, CronName+"-"+ns, v1.DeleteOptions{})
+}
+
+func (pruner *Pruner) removedFromTektonConfig(prune v1alpha1.Prune) bool {
+	if len(prune.Resources) == 0 || prune.Schedule == "" {
+		return true
+	}
+	return false
+}
+
+func (pruner *Pruner) removedNSAnnotation() ([]string, bool) {
+	var namespaces []string
+	cronJobs, err := pruner.kc.BatchV1beta1().CronJobs("").List(pruner.context, v1.ListOptions{})
+	if err != nil {
+		return nil, false
+	}
```

Comment:
why don't we return error..  if no cronjobs are returned, we can check for  errorNotFound and return nil. [example](https://github.com/tektoncd/operator/blob/a3d280298cf08d090da741e2f6d0ee17b09e9d5f/pkg/reconciler/shared/tektonconfig/tektonconfig.go#L125)

Reply:
Good catch, changes will appear ones pr is updated.